development-vm/replication/close_and_delete_old_indices.rb

```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/env ruby
+
+require 'net/http'
+require 'json'
+require 'uri'
+
+valid_indices = ARGV[0].split(',')
+
+list_uri = URI.parse('http://localhost:9200/_alias')
+
+Net::HTTP.start(list_uri.host, list_uri.port) do |http|
+  resp = http.get(list_uri.path)
+  data = JSON.parse(resp.body)
+
+  used_aliases = valid_indices.flat_map { |index_name| data[index_name]['aliases'].keys }
```

Comment:
I think you need to change this line to account for when there's no `data[index_name]`

```ruby
  used_aliases = valid_indices.flat_map do |index_name|
    next unless data[index_name]
    data[index_name]['aliases'].keys
  end
```